src/function/gds/gds_task.cpp

```diff
@@ -53,10 +53,12 @@ void VertexComputeTask::run() {
     while (sharedState->morselDispatcher.getNextRangeMorsel(frontierMorsel)) {
         while (frontierMorsel.hasNextOffset()) {
```

Comment:
I see two options. 
Option 1 (General early stopping logic):  Either have a general breaking early optimization similar to edgeCompute() but not exactly the same based on implementing some abstract functions. 
Option 2 (Only algorithm-specific early stopping logic): Do not have any early stopping logic here, so this code does not change and then push all early stopping logics to the algorithms, which can implement early stopping inside their vertexCompute functions. This means, we always call vertexCompute() on each node, but those vertexCompute() functions would just return without doing any work.

I'm fine with either approach frankly. You can choose the simpler one. But I would probably not do something in between, like this one, which has some logic of general early stopping implemented in algorithm-agnostic parts of the code like gds_task.cpp, plus some algorithm-specific code (but not abstracted away cleanly).

Reply:
Took the second approach.